src/drivers/ports/avr/watchdog_port.i

```diff
@@ -33,7 +33,29 @@
 int watchdog_port_start_ms(int timeout,
                            watchdog_isr_fn_t on_interrupt)
 {
-    wdt_enable(timeout);
+    uint8_t prescaler;
+
+    /*
+     * http://www.nongnu.org/avr-libc/user-manual/group__avr__watchdog.html
+     *
+     * Possible timeout values are:
+     * 15ms => prescaler 0
+     * 30ms => prescaler 1
+     * 60ms => prescaler 2
+     * 120ms => prescaler 3
+     * 250ms => prescaler 4
+     * 500ms => prescaler 5
+     * 1s => prescaler 6
+     * 2s => prescaler 7
+     */
+    timeout /= 15;
+    prescaler = 0;
+    while ((1 << prescaler) < timeout) {
+        prescaler++;
+        if (prescaler == 7) break;
+    }
+
+    wdt_enable(prescaler);
```

Comment:
The timeout values are not 15, 30, 60, ... according to the MCU reference manual, but instead 16, 32, 64, ..., so I think you should divide by 16 instead of 15.

Also, the function gives a prescaler of 0 for a timeout of 29 ms. I expect the function to set a watchdog timeout of given timeout or longer, not shorter. The function seems to round down to closest prescaler value.